pkg/porter/dependencies.go

```diff
@@ -248,22 +249,31 @@ func (e *dependencyExecutioner) executeDependency(dep *queuedDependency, parentA
 		// For now, assume it's okay to give the dependency the same credentials as the parent
 		CredentialIdentifiers: parentArgs.CredentialIdentifiers,
 	}
+
+	var executeErrs error
 	fmt.Fprintf(e.Out, "Executing dependency %s...\n", dep.Alias)
 	err := e.CNAB.Execute(depArgs)
 	if err != nil {
-		return errors.Wrapf(err, "error executing dependency %s", dep.Alias)
-	}
+		executeErrs = multierror.Append(executeErrs, errors.Wrapf(err, "error executing dependency %s", dep.Alias))
 
-	// If action is uninstall, no claim will exist
-	if parentArgs.Action != claim.ActionUninstall {
-		// Collect expected outputs via claim
-		outputs, err := e.Claims.ReadLastOutputs(depArgs.Installation)
-		if err != nil {
-			return err
+		// Handle errors when/if the action is uninstall
+		executeErrs = e.parentOpts.handleUninstallErrs(e.Out, executeErrs)
+		if executeErrs != nil {
+			return executeErrs
 		}
+	}
 
-		dep.outputs = outputs
+	if e.parentOpts.shouldDelete() {
```

Comment:
I think you put the `UninstallDeleteOptions` on the generic `BundleLifecycleOptions` so that you could make this check here, right?

Instead, since I'm asking further down to move that to the `UninstallOptions` type would be to do a type check

```go
if uninstOpts, ok := e.parentOpts.(UninstallOptions); ok {
...
}
```

Right now `parentOpts` is just a `BundleLifecycleOptions` but we could make it an interface (pseudo code, not tested, but something like below), so that we could cast and get at the original options as needed while still having access to the other data that we were using on the parentOpts the rest of the time.

What do you think?

```golang
type BundleAction interface {
    // Implemented for free because of the embedded BundleLifecycleOpts
    ToActionArgs()

    // Let us grab and get direct access to the bundle opts fields like we need in some funcs
    GetBundleLifecycleOptions()
}
```

Reply:
Great suggestion!  Thank you for providing the blueprint on this approach.  Here's what I've updated: https://github.com/deislabs/porter/pull/1189/commits/4b2b64746636e6997ee1f3dae8c87953c8b0cddc

How does it look?